client/src/pages/Actions.ts

```diff
@@ -1,7 +1,38 @@
-import { STORE_API, USER_API } from "src/constants";
+import {
+  STORE_API,
+  STORE_LIST_API,
+  USER_API,
+  ITEM_API,
+  ITEM_LIST_API,
+  DEFAULT_STORE_RADIUS,
+  PLACES_RADIUS_METERS,
+  PLACES_TYPES,
+  GOOGLE_MAP_PLACEHOLDER_ID,
+  GOOGLE_PLACES_SCRIPT_ID,
+} from "src/constants";
 import { fetchJson, extractIdentityToken } from "src/utils";
-import { isWeb, microapps } from "src/config";
-import { IdentityToken } from "src/interfaces";
+import { isWeb, google, microapps } from "src/config";
+import { IdentityToken, GeoLocation, emptyGeoLocation } from "src/interfaces";
+
+// Load up Google Maps Places API Service
+let map: any;
+let service: any;
+
+const loadGooglePlacesService = () => {
+  map = new google.maps.Map(document.getElementById(GOOGLE_MAP_PLACEHOLDER_ID));
+  service = new google.maps.places.PlacesService(map);
+};
+
+// Depending on when Actions.ts is loaded, we may not have retrieved
+// google Places API Script yet. If so, watch <script> for load prior to init
+if (google) {
+  loadGooglePlacesService();
+} else {
+  const googleScript = document.getElementById(GOOGLE_PLACES_SCRIPT_ID);
+  if (googleScript) {
+    googleScript.addEventListener("load", loadGooglePlacesService);
+  }
+}
```

Comment:
i'm assuming that the intention of this code is to run it everytime Actions.ts is used in any js file?

configuration of global variables for scripts should be done inside a startUp function

Reply:
I might be wrong here but I was working off the assumption that imports are cached and this will only ever be run once when `Actions.ts` is first requested. Then this will fire off a listener only once to update variables if scripts have yet to be loaded.
But yes, valid point here, this is not quite good design, will rethink how startup initialization can be done here